pkg/kubelet/winstats/perfcounter_nodestats.go

```diff
@@ -140,13 +144,25 @@ func (p *perfCounterNodeStatsClient) getMachineInfo() (*cadvisorapi.MachineInfo,
 	}
 
 	return &cadvisorapi.MachineInfo{
-		NumCores:       runtime.NumCPU(),
+		NumCores:       processorCount(),
 		MemoryCapacity: p.nodeInfo.memoryPhysicalCapacityBytes,
 		MachineID:      hostname,
 		SystemUUID:     systemUUID,
 	}, nil
 }
 
+func processorCount() int {
+	if amount := getActiveProcessorCount(allProcessorGroups); amount != 0 {
```

Comment:
nit: it would be nice to have comment here on why it fails for some scenarios for future us

Reply:
To be honest, I don’t know. I just find  a way to fix this bug in https://github.com/moby/moby/issues/38935#issuecomment-744638345  and copy code to k8s .

I need some help from people who know windows.  Thank you for reviewing this PR.

Reply:
Thanks!

> This server has four cpus, each with 18 cores and two logical threads in the core. So a total of 72 cores, 144 threads. the result obtained by kubelet is 36 logical CPUs. When I adopted this PR code, I got the number in the picture above, which is 144.

> As was mentioned above, many of the Windows processor affinity functions will only return the information for a single Processor Group. Since a single group can only hold 64 logical processors, this means when there are more they will be divided into multiple groups (as seen above, where on a 72 LP system there are 2 groups of 36 LPs).

This sounds like what I was wondering about.  I don't know that we can cover such a use case in our tests.

Reply:
could you add a comment about the issue with logical processors and give attribution in a comment to hcsshim?

Reply:
Thanks for your suggestion. I added some comment for this issue. I don't know if it can be written like this. Please review it again.

Reply:
Thanks, that helps to understand the issue and fix.  two really minor wording changes and we are good!